# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/gtest/mediapipeline_unittest.cpp
# Commit: c54cb3c10992
# Full Hash: c54cb3c109922bc15fb6d0ca67e5f2a05980c1a5
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 2
# Description:
#   Bug 1493613 - Move MediaStream control from DOMMediaStream to MediaStreamTrack. r=padenot
#   
#   This is inherently large, because modifying these bits of DOMMediaStream and
#   MediaStreamTrack affects all consumers and producers of all DOMMediaStreams and
#   MediaStreamTracks.
# ==============================================================================

diff -r 0b03dd9d20ac -r c54cb3c10992 media/webrtc/signaling/gtest/mediapipeline_unittest.cpp
--- a/media/webrtc/signaling/gtest/mediapipeline_unittest.cpp	Wed Jul 31 09:35:06 2019 +0000
+++ b/media/webrtc/signaling/gtest/mediapipeline_unittest.cpp	Wed Jul 31 07:58:17 2019 +0000
@@ -11,7 +11,6 @@
 
 #include "AudioSegment.h"
 #include "AudioStreamTrack.h"
-#include "DOMMediaStream.h"
 #include "mozilla/Mutex.h"
 #include "mozilla/RefPtr.h"
 #include "MediaPipeline.h"
@@ -38,19 +37,6 @@
 
 namespace {
 
-class FakeSourceMediaStream : public mozilla::SourceMediaStream {
- public:
-  FakeSourceMediaStream() : SourceMediaStream() {}
-
-  virtual ~FakeSourceMediaStream() override { mMainThreadDestroyed = true; }
-
-  virtual StreamTime AppendToTrack(
-      TrackID aID, MediaSegment* aSegment,
-      MediaSegment* aRawSegment = nullptr) override {
-    return aSegment->GetDuration();
-  }
-};
-
 class FakeMediaStreamTrackSource : public mozilla::dom::MediaStreamTrackSource {
  public:
   FakeMediaStreamTrackSource() : MediaStreamTrackSource(nullptr, nsString()) {}
@@ -69,8 +55,7 @@
 class FakeAudioStreamTrack : public mozilla::dom::AudioStreamTrack {
  public:
   FakeAudioStreamTrack()
-      : AudioStreamTrack(new DOMMediaStream(nullptr), 0, 1,
-                         new FakeMediaStreamTrackSource()),
+      : AudioStreamTrack(nullptr, nullptr, 0, new FakeMediaStreamTrackSource()),
         mMutex("Fake AudioStreamTrack"),
         mStop(false),
         mCount(0) {